### DISCUSSION TITLE
How do I import data from one branch to another?

### DISCUSSION THREAD

**Discussion by jonrcrowell** [planetscale/discussion#141] [Q&A] [answered]
My main branch has the exact same schema as on of my dev branches but is empty. The dev branch has data that I want in main. How can I load the data in dev into main?

I've tried using MySQL Workbench Migration Wizard, as well as the Table Data Export and Import Wizards without any luck.

**Comment by jonrcrowell** [comment]
Hmmm. The MySQL Wizards all reported failure but I just checked the tables and most of them are populated. I'll check on the remaining tables and see if I just need to be patient and wait for an async process to complete.

**Comment by jgreet** [comment, marked as answer]
Sounds like you may have figured it out but just in case, or for anyone else in the same situation, I can think of a couple other options:

In PlanetScale, assuming you don't need anything from the existing **main** branch:
- In database settings, make **dev** the default branch
- Delete the **main** branch
- [Restore a backup](https://docs.planetscale.com/concepts/back-up-and-restore#restore-from-a-backup) of **dev** to a new branch named **main**
- Make this new **main** the default branch

You could skip the branch deletion if you're OK with giving your new default branch another name like **prod**.

Using mysqldump:
- Dump the data from **dev**:
   ```
   mysqldump --no-create-info --skip-disable-keys --user=<username> --password=<password> --host=<host> --result-file=<filename.sql> <database>
   ```
- [Run the inserts from your result-file](https://dev.mysql.com/doc/refman/8.0/en/mysql-batch-commands.html) on **main**:
   `pscale shell <database> main` and `source <filename.sql>`

**Reply by Nesquik** [reply]
Hi @jgreet,
is this procedure still possible? I'm trying to move data over branches aswell after deploying changes through a request from one branch to the other.

I've promoted my develop branch like you suggested to default and made a manual backup out of it. 
After that, I can't delete the main branch straight out since it has a child branch, which is, sure enough, develop.

![Schermata 2022-05-07 alle 16 44 18](https://user-images.githubusercontent.com/216658/167259413-4af0a21a-502e-4f6f-a1ac-b6f23264f6be.png)

If I were to delete develop and then main, I wouldn't be able to restore the manual backup as it'd delete that aswell, or would I?

I've also gone ahead and tried your second option but ran into this:
```
mysqldump: Couldn't execute 'SELECT /*!40001 SQL_NO_CACHE */ * FROM `_PersonToWikipediaCategory`': vttablet: rpc error: code = Aborted desc = Row count exceeded 100000 (errno 10001) (sqlstate HY000) (CallerID: planetscale-admin): Sql: "select * from _PersonToWikipediaCategory", BindVars: {} (10001)
```
which looks like would be the same issue as https://github.com/planetscale/discussion/discussions/190

Thanks for the help

**Reply by jgreet** [reply]
I believe the first option will still work but now you have to promote **develop** to [production status](https://docs.planetscale.com/concepts/branching#promote-a-branch-to-production), in addition to changing the default branch, before you can delete the original **main**.

If you need to dump a table with more than 100k rows try using the dumper built in to [pscale](https://github.com/planetscale/cli) instead:
```
pscale database dump <database> <branch>
```

If you needed more than 100k rows returned in a normal session you could [switch to the OLAP workmode](https://vitess.io/docs/13.0/reference/compatibility/mysql-compatibility/#olap-workload) but I don't think there's a way to specify it for `mysqldump`.

**Reply by end0cr1ne** [reply]
Does switching to OLAP with: set workload=olap; work for anyone?
